libmscore/unrollrepeats.cpp

```diff
@@ -0,0 +1,352 @@
+                        if (ns) ns->remove(ns->element(staffIdx));
+                  }
+            
+            // Check if time signature needs to be changed
+            if (!((*ostaff->timeSig(otick))==(*staff->timeSig(ctick)))) {
+                  Segment* ns = firstAppendedMeasure->undoGetSegment(SegmentType::TimeSig, ctick);
+                  TimeSig * nsig = new TimeSig(*(ostaff->timeSig(otick)));
+
+                  nsig->setScore(score);
+                  nsig->setTrack(staffIdx * VOICES);
+                  ns->add(nsig);
+                  //score->undoAddElement(nsig);
+                  }
+            // Check if a time signature is present but is spurious (i.e. no actual change)
+            else if (staff->currentTimeSigTick(ctick)==ctick &&
+                     (*staff->timeSig(ctick-1))==(*ostaff->timeSig(otick))) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::TimeSig);
+                        if (ns) ns->remove(ns->element(staffIdx));
+                  }
+
+            // Check if clef signature needs to be changed
+            if (!(ostaff->clef(otick)==staff->clef(ctick))) {
+                  score->undoChangeClef(staff,firstAppendedMeasure,ostaff->clef(otick));
+                  }
+            // Check if a clef change is present but is spurious (i.e. no actual change)
+            else if (staff->currentClefTick(ctick)==ctick &&
+                     staff->clef(ctick-1)==ostaff->clef(otick)) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::Clef);
+                        if (!ns) ns = firstAppendedMeasure->first(SegmentType::HeaderClef);
+                        if (ns) ns->remove(ns->element(staffIdx));
+                  }
+            }
+
+      // Check if section starts with apick-up measure to be merged with end of previous section
+      Measure * cm = firstAppendedMeasure, * pm = cm->prevMeasure();
+      if (pm->timesig() == cm->timesig() && 
+          pm->len() + cm->len() == cm->timesig()) {
+            qDebug() << "Detected a pre-measure that should be joined with previous measure at " << tickOfAppend;
+            score->cmdJoinMeasure(pm,cm);
+            }
+      
+
+      // clone the spanners (only in the range currently copied)
+      auto ospans = original->spanner();
+      auto lb = ospans.lower_bound(btick), ub = ospans.upper_bound(etick);
+      for (auto sp=lb;sp!=ub;sp++) {
+            Spanner* spanner = sp->second;
+            
+            if (spanner->tick2()>etick) continue; // Map is by tick() so this can still happen in theory...
+
+            Spanner * ns = toSpanner(spanner->clone());
+            ns->setScore(score);
+            ns->setParent(0);
+            ns->setTick(spanner->tick() - btick + tickOfAppend);
+            ns->setTick2(spanner->tick2() - btick + tickOfAppend);
+            if (ns->isSlur()) {
+                  // set start/end element for slur
+                  ns->setStartElement(0);
```

Comment:
@dmitrio95 copy-pasted from appendScore in score.cpp . I don't really understand your proposal though, so maybe you can elaborate a bit?.

Reply:
Ah, that code is from another part of libmscore. Then it is interesting why it is done so here.
I mean, if you just write
```
ns->computeStartElement();
ns->computeEndElement();
```
then there would probably no need for all that code below to set start and end element properly. In fact, I am not even sure that start and end elements are needed at all at this stage, perhaps only resetting them to `nullptr` would be enough.

And it is very curious that this all is done only for slurs while other spanners can also have start/end elements, it would be logical to do that for all spanners instead.

So I would propose to check whether replacing all that `if` block with
```
ns->setStartElement(nullptr);
ns->setEndElement(nullptr);
// or, alternatively
ns->computeStartElement();
ns->computeEndElement();
```
would work properly. If so I believe it would be preferable to use one of these options rather than preserving this chunk of code.

Reply:
@dmitrio95 This seems to have worked. Thank you for suggesting this.